dev/source/docs/building-px4-with-qtcreator.rst

```diff
@@ -0,0 +1,233 @@
+   ``Build & Run`` tab page. 	
+
+#. Click ``Manage Kits`` in the topleft corner.
+
+#. First click on the ``Compilers`` tab page and then "Add" on the right hand side of the compilers list.
+   Choose an easily recognisable name for your compiler and make sure the Compiler and Make path are 
+   referring to the executables of the PX4 toolchain (Windows) or the downloaded gcc-arm cross-compiler
+   (Linux). Also choose "GCC" as the Error parser.
+   
+   .. image:: ../images/QtCreator_ManageCompilers.png
+	 :target: ../_images/QtCreator_ManageCompilers.png     
+
+#. Then click on the ``Kits`` tab page. Click ``Add`` on the right hand side.
+
+   Choose an easily recognisable name for your build kit and make sure you fill in the
+   proper compiler (the one you just added) and the debugger inside the PX4 toolchain.
+
+   .. image:: ../images/QtCreator_ManageKits.png
+	  :target: ../_images/QtCreator_ManageKits.png   
+
+#. Click apply.
+
+#. Back on the ``Projects`` page, click ``Add Kit`` and choose the Build Kit you just added.
+
+#. Now you have one "Build Configuration" called "Default". You can make as many Build Configurations
+   as you want, but we'll take the ArduCopter build as an example for now. 
+   Next to ``Edit build configuration``, click on ``Add`` and choose ``Clone Selected`` in the 
+   drop down menu. Pick a name (e.g. "Copter").
+   
+#. Click on the ``Details`` of the ``Build Steps`` and type "px4-v2 -j2" as ``Make arguments``. 
+   Deselect the ``Targets``.
+
+#. Click on the ``Details`` of the ``Clean Steps`` and type "px4-clean" as ``Make arguments``.
+   Deselect the ``Targets``.
+
+   .. image:: ../images/QtCreator_Target_Project_Settings.png
+      :target: ../_images/QtCreator_Target_Project_Settings.png   	
+	
+#. You can make other build configurations for e.g. ArduPlane in the same way. You can quickly switch
+   between "Build Configurations" by clicking the logo just above the ``Run`` icon (the green arrow) on 
+   the left pane. 	
+	
+#. You can now remove the MSVC or standard GCC build kit (click on the down arrow on the kit itself and 
+   choose ``Remove Kit``)
+   
+#. You're now ready to build the code. Click on ``Edit`` in the left pane to edit the code and browse
+   through the project. Click ``Build project-name`` in the ``Build`` menu (or Ctrl+B) to build the code.    
+
+Qt Creator editor settings
+==========================		
+#. It is handy to set the tab and indentation sizes to be used while editing. Click on the ``Tools`` menu
+   and choose ``Options``. Subsequently, pick the ``Text Editor`` view and then the ``Behaviour`` tab page.
+   You can set the tab policy (e.g. spaces only), the size of a tab and indentations.
+
+#. Other interesting settings can be found in the ``C++`` view in the same ``Options`` dialog. You can define
+   how specific parts of your code will be aligned (e.g. assignments, switch/cases, etc.)
+
+Doxygen
```

Comment:
Thank for the explanation. I suggest that you simply add a sentence at the top explaining that "In order to comply with the coding guidelines <link>, you will need to provide docmentation in Doxygen format." 

That way it is clear why this information is relevant/part of the story. Make sense?